editoast/src/views/infra/mod.rs

```diff
@@ -801,6 +807,66 @@ pub async fn fetch_all_infra_states(
         .collect())
 }
 
+#[derive(Serialize, Deserialize, ToSchema)]
+struct WaypointsForm {
+    waypoints: Vec<OperationalPointReference>,
+}
+
+/// Retrieve operational points from operational point references
+#[utoipa::path(
+    get, path = "",
+    tag = "infra",
+    params(InfraIdParam),
+    request_body = inline(WaypointsForm),
+    responses(
+        (status = 200, description = "Operational points related to operational point references", body = Vec<Vec<OperationalPoint>>)
+    ),
+)]
+async fn match_operational_points(
+    State(AppState { db_pool, .. }): State<AppState>,
+    Extension(auth): AuthenticationExt,
+    Path(InfraIdParam { infra_id }): Path<InfraIdParam>,
+    Json(waypoints_input): Json<WaypointsForm>,
+) -> Result<Json<Vec<Vec<OperationalPoint>>>> {
+    let authorized = auth
+        .check_roles([Role::OperationalStudies].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+    let mut operational_points: Vec<Vec<OperationalPoint>> = vec![];
+    let conn = &mut db_pool.get().await?;
+    for waypoint in waypoints_input.waypoints {
+        operational_points.push(match waypoint.reference {
+            OperationalPointIdentifier::OperationalPointId {
+                operational_point: op_id,
+            } => retrieve_op_from_ids(conn, infra_id, &[op_id.0])
```

Comment:
nitpick: not sure of the value of renaming an existing field here. Maybe you can just do the following?

```suggestion
            OperationalPointIdentifier::OperationalPointId {
                operational_point,
            } => retrieve_op_from_ids(conn, infra_id, &[operational_point.0])
```